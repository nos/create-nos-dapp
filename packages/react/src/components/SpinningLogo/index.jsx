import React from "react";
import injectSheet from "react-jss";
import PropTypes from "prop-types";

import nosLogo from "./../../assets/logo.png";

const styles = {
  logoContainer: {
    display: "block",
    maxWidth: "200px",
    margin: "32px auto",
    position: "relative"
  },
  nosLogo: {
    animation: "nOS-logo-spin infinite 3s linear",
    width: "100%"
  },
  "@keyframes nOS-logo-spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(360deg)" }
  }
};

const SpinningLogo = ({ classes }) => (
  <div className={classes.logoContainer}>
    <img src={nosLogo} className={classes.nosLogo} alt="nOS logo" key="nosLogo" />
  </div>
);

SpinningLogo.propTypes = {
  classes: PropTypes.object.isRequired
};

export default injectSheet(styles)(SpinningLogo);
