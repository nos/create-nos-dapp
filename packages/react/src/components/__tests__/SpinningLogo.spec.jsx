import React from "react";
import { render } from "enzyme";
import SpinningLogo from "./../SpinningLogo";

describe("Components", () => {
  describe("SpinningLogo", () => {
    test("should render", () => {
      const wrapper = render(<SpinningLogo />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
