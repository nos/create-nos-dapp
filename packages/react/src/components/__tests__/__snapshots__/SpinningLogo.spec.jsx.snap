// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components SpinningLogo should render 1`] = `
<div
  class="SpinningLogo-logoContainer-0-1-1"
>
  <img
    alt="nOS logo"
    class="SpinningLogo-nosLogo-0-1-2"
    src="test-file-stub"
  />
</div>
`;
